vividus-plugin-web-app-playwright/src/main/java/org/vividus/ui/web/playwright/BrowserContextProvider.java

```diff
@@ -90,39 +96,14 @@ public void closeCurrentContext()
                 });
     }
 
-    // Impossible because 'destroy' is invoked by Spring in thread-safe way
-    @SuppressFBWarnings("IS2_INCONSISTENT_SYNC")
-    public void destroy()
+    public void closePlaywrightContext()
     {
-        Optional.ofNullable(playwrightContext).ifPresent(context -> {
-            context.browser().close();
-            context.playwright().close();
-        });
-    }
-
-    // This implementation of DC Locking algorithm is correct,
-    // check: https://shipilev.net/blog/2014/safe-public-construction/
-    @SuppressFBWarnings({ "IS2_INCONSISTENT_SYNC", "DC_DOUBLECHECK" })
-    // Playwright and Browser instances are closed in 'destroy' method
-    @SuppressWarnings("PMD.CloseResource")
-    private PlaywrightContext getPlaywrightContext()
-    {
-        PlaywrightContext pc = playwrightContext;
-        if (pc == null)
-        {
-            synchronized (this)
-            {
-                pc = playwrightContext;
-                if (pc == null)
-                {
-                    Playwright playwright = Playwright.create();
-                    Browser browser = browserType.launchBrowser(playwright, launchOptions);
-                    pc = new PlaywrightContext(playwright, browser);
-                    playwrightContext = pc;
-                }
-            }
-        }
-        return pc;
```

Comment:
what is the root cause of these changes?

Reply:
The root cause is that we can't use the same playwright instance for all tests.
https://github.com/microsoft/playwright-java/issues/1184
https://playwright.dev/java/docs/multithreading